tests/simulation/lazy_tensor_op_test.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (C) 2020-2021 - George Watkins, Alex Nguyen, Varun Seshadri
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+# USA
+from typing import List
+
+import pytest
+import qiskit.opflow as qkop
+
+from lsqecc.simulation.lazy_tensor_op import (
+    LazyTensorOp,
+    LazyTensorOpsNotMatchingException,
+)
+from lsqecc.simulation.qiskit_opflow_utils import to_vector
+from tests.simulation.numpy_matrix_assertions import (
+    assert_eq_numpy_matrices,
+    assert_eq_numpy_vectors,
+)
+
+
+class TestLazyTensorOp:
+    @pytest.mark.parametrize(
+        "first, second, expected",
+        [
+            ([qkop.Zero], [qkop.X], [qkop.One]),
+            ([qkop.Plus], [qkop.Z], [qkop.Minus]),
+            ([qkop.X], [qkop.Z], [1j * qkop.Y]),
+            ([qkop.X, qkop.I ^ qkop.Z], [qkop.I, qkop.Y ^ qkop.I], [qkop.X, qkop.Y ^ qkop.Z]),
```

Comment:
is it called Lazy because the Hilbert space which the tensor applies is not explicitly specified?

Reply:
As discussed over zoom, it's lazy in the sense that it is lazy evaluated. Added comment in code